src/pybind/mgr/dashboard_v2/controllers/rbd_mirroring.py

```diff
@@ -0,0 +1,337 @@
+        def get_daemons():
+            daemons = []
+            for server in self.mgr.list_servers():
+                for service in server['services']:
+                    if service['type'] == 'rbd-mirror':
+                        id = service['id']  # pylint: disable=W0622
+                        metadata = self.mgr.get_metadata('rbd-mirror', id)
+                        status = self.mgr.get_daemon_status('rbd-mirror', id)
+                        try:
+                            status = json.loads(status['json'])
+                        except ValueError:
+                            status = {}
+
+                        instance_id = metadata['instance_id']
+                        if id == instance_id:
+                            # new version that supports per-cluster leader elections
+                            id = metadata['id']
+
+                        # extract per-daemon service data and health
+                        daemon = {
+                            'id': id,
+                            'instance_id': instance_id,
+                            'version': metadata['ceph_version'],
+                            'server_hostname': server['hostname'],
+                            'service': service,
+                            'server': server,
+                            'metadata': metadata,
+                            'status': status
+                        }
+                        daemon = dict(daemon, **get_daemon_health(daemon))
+                        daemons.append(daemon)
+
+            return sorted(daemons, key=lambda k: k['instance_id'])
+
+        def get_daemon_health(daemon):
+            health = {
+                'health_color': 'info',
+                'health': 'Unknown'
+            }
+            for _, pool_data in daemon['status'].items():
+                if (health['health'] != 'error' and
+                        [k for k, v in pool_data.get('callouts', {}).items()
+                         if v['level'] == 'error']):
+                    health = {
+                        'health_color': 'error',
+                        'health': 'Error'
+                    }
+                elif (health['health'] != 'error' and
+                      [k for k, v in pool_data.get('callouts', {}).items()
+                       if v['level'] == 'warning']):
+                    health = {
+                        'health_color': 'warning',
+                        'health': 'Warning'
+                    }
+                elif health['health_color'] == 'info':
+                    health = {
+                        'health_color': 'success',
+                        'health': 'OK'
+                    }
+            return health
+
+        def get_pools(daemons):  # pylint: disable=R0912, R0915
+            Dashboard.mgr = self.mgr
+            dashboard = Dashboard()
+            # TODO: move out of dashboard
+            _, pool_names = dashboard._rbd_pool_ls()  # pylint: disable=W0212
```

Comment:
Yes, must not do this. If we need common functionality in two different controllers then we should create a "service" class that can provide that data for both controllers.

Reply:
I'm not completely sure if I understand you correctly. Do you mean I should move it away to a service class or we will move it at a later point?

Reply:
@s0nea I just created PR #127 to address this issue. In PR #127 is added a function to get the list of rbd pools. This function is already being used by the dashboard function you are calling. So instead of trying to use the code of other controller, as you are doing, you can get the list of rbd pools with the following code:

`CephService.get_pool_list('rbd')`